backport: bitcoin#18275, #20220, #20305, #20410, #20426, #20573, #21083, #21201, #21786, #21787 (fee backports)

knst opened the pull request:
## Issue being fixed or feature implemented
Just regular backports from bitcoin v0.21, v22; mostly wallet+fee related

## What was done?
 - bitcoin/bitcoin#18275
 - bitcoin/bitcoin#20220
 - bitcoin/bitcoin#20305
 - bitcoin/bitcoin#20426
 - bitcoin/bitcoin#20410
 - bitcoin/bitcoin#20573
 - bitcoin/bitcoin#21201
 - bitcoin/bitcoin#21083
 - bitcoin/bitcoin#21787

## How Has This Been Tested?
Run unit and functional tests

## Breaking Changes
Some wallet rpc (sendtoaddress, sendmany, send) have a new argument `fee_rate` which is inserted before `verbose`.
Release notes will be provided in a new PR once scope of backports and fixes in this PR is finalized by merging it to develop/

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

UdjinM6 commented on src/coinjoin/util.cpp:
```diff
@@ -116,7 +116,7 @@ CTransactionBuilder::CTransactionBuilder(std::shared_ptr<CWallet> pwalletIn, con
     coinControl.m_discard_feerate = ::GetDiscardRate(*pwallet);
     // Generate a feerate which will be used by calculations of this class and also by CWallet::CreateTransaction
-    coinControl.m_feerate = std::max(pwallet->chain().estimateSmartFee(int(pwallet->m_confirm_target), true, nullptr), pwallet->m_pay_tx_fee);
+    coinControl.m_confirm_target = pwallet->m_confirm_target;
```
coinjoin test is still broken

knst commented:
> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/9dad07e5fa94c4358ee9df289b735c090eec62b6..61a1b782403e5d185c218803092ee37cd8e8deb4) the bp-v21-p28 branch from[ 9dad07e ](https://github.com/dashpay/dash/commit/9dad07e5fa94c4358ee9df289b735c090eec62b6)to 61a1b78

addressed comments from PR. Still draft so far as CJ issue is not addressed yet.

UdjinM6 commented:
Pls consider 0206704a0b + dropping b3d540b. The fix tldr is basically switching from fee estimation to cmd-line/cfg params only for CJ. It should work just fine since 99.9999% of the time fees are at the base level and no estimation is needed at all.

UdjinM6 commented on src/policy/feerate.h:
```diff
@@ -39,7 +39,16 @@ class CFeeRate
+     *  @param[in] nFeePaid  CAmount fee rate to construct with
+     *  @param[in] nBytes    size_t bytes (units) to construct with
+     *  @returns   fee rate
+     */
```
still not fixed, check https://github.com/bitcoin/bitcoin/pull/21848/files#diff-1e929c95956c19f3b5bad7269f871fb1d4f3ee0ba0856df468f0fb9c5032a920R42-R48

PastaPastaPasta reviewed (approved):
utACK d6946aaba1550f1f2e81a965f8ec4be1604e1f3a
